source/MaterialXContrib/MaterialXNode/MaterialXData.cpp

```diff
@@ -111,15 +139,9 @@ void MaterialXData::generateXml()
     // Set up generator context. For shaders use FIS environment lookup,
     // but disable this for textures to avoid additional unneeded XML parameter
     // generation.
-    _genContext.registerSourceCodeSearchPath(_librarySearchPath);
-    if (shaderRef)
-    {
-        _genContext.getOptions().hwSpecularEnvironmentMethod = mx::SPECULAR_ENVIRONMENT_FIS;
-    }
-    else
-    {
-        _genContext.getOptions().hwSpecularEnvironmentMethod = mx::SPECULAR_ENVIRONMENT_NONE;
-    }
+    _genContext.getOptions().hwSpecularEnvironmentMethod =
```

Comment:
Do you think we could make the generator and its context local to the generateFragment() function. We won't need them once the shader has been generated.

Reply:
Yes.

Reply:
Great, I'll make this change.